src/shared/cloudformation/templateRegistry.ts

```diff
@@ -125,21 +128,117 @@ export class CloudFormationTemplateRegistry {
 }
 
 /**
- * Helper function that returns an map of resource names to CloudFormation.Resource objects.
- * Unlike a CloudFormation.TemplateResources object, all resources in this array are guaranteed to be defined.
- * @param templateDatum TemplateDatum object to extract resources from
+ * Gets resources and additional metadata for resources tied to a filepath and handler.
+ * Checks all registered templates by default; otherwise can operate on a subset TemplateDatum[]
+ * @param filepath Handler file's path
+ * @param handler Handler function from aforementioned file
+ * @param unfilteredTemplates Array containing TemplateDatum objects to filter
  */
-export function getResourcesFromTemplateDatum(
-    templateDatum: TemplateDatum,
-    resourceTypes: CloudFormation.ResourceType[] | undefined = undefined
-): Map<string, CloudFormation.Resource> {
-    const map = new Map<string, CloudFormation.Resource>()
-    for (const resourceKey of Object.keys(templateDatum.template.Resources!)) {
-        const resource = templateDatum.template.Resources![resourceKey]
-        if (resource && (resourceTypes === undefined || resourceTypes.includes(resource.Type))) {
-            map.set(resourceKey, resource)
+export function getResourcesAssociatedWithHandler(
+    filepath: string,
+    handler: string,
+    unfilteredTemplates: TemplateDatum[] = CloudFormationTemplateRegistry.getRegistry().registeredTemplates
+): { templateDatum: TemplateDatum; name: string; resourceData: CloudFormation.Resource }[] {
+    // TODO: Array.flat and Array.flatMap not introduced until >= Node11.x -- migrate when VS Code updates Node ver
+    return unfilteredTemplates
+        .map(templateDatum => {
+            return getResourcesAssociatedWithHandlerFromTemplateDatum(filepath, handler, templateDatum).map(
+                resource => {
+                    return {
+                        ...resource,
+                        templateDatum,
+                    }
+                }
+            )
+        })
+        .reduce((acc, cur) => [...acc, ...cur])
+}
+
+/**
+ * Returns an array of Cloudformation Resources in a TemplateDatum that is tied to the filepath and handler given.
+ * @param filepath Handler file's path
+ * @param handler Handler function from aforementioned file
+ * @param templateDatum TemplateDatum object to search through
+ */
+export function getResourcesAssociatedWithHandlerFromTemplateDatum(
```

Comment:
ShorterNameNoLossOfClarityIncreasedReadability:

```suggestion
export function getResourcesForHandlerFromTemplateDatum(
```